# 인사

안녕하세요 ~~ 25기 여러분 반갑습니다 ~~  

드디어 정규세션이 시작되었네요.  

뭘 과제로 내볼까 고민하다가, 요즘 데이터 분석 채용 공고 자격 요건에 필수로 들어있는.... A/B TEST가 좋겠다!!고 생각했습니다.  

데이터 분석쪽이 아니더라도, 다양한 도메인에서 쓸 수 있는 방법이라고 생각해요.


A/B TEST의 개념부터 기초적인 과정을 이해해볼 수 있게 내봤어요. 근데 설명해야할 게 많다보니 텍스트가 너무 많아졌네요.

읽기 힘들어도 ㅎㅎ... 열심히 해주시고, 질문 사항은 저에게 연락 남겨주세요.  

참고로, 정규세션 발표 외의 내용을 구글링하고 공부해야만 풀 수 있는 문제들이 포함되어 있습니다.

제 이름... 아시죠...? 24기 박현서입니다....



<br>

# Q1. A/B TEST란 무엇일까요?   
또한, A/B TEST시 유의할 사항들을 초두효과를 포함해 기술해주세요.

정규세션에서 발표한 내용 외에 추가적으로 공부한 개념을 작성하셔도 됩니다.
정의와 함께 예시 하나만 들어주세요 ~~

## A1. 답안 작성  
답변:A/B 테스트는 웹/앱의 두 버전( 기존, 변경안)을 사용자에게 무작위로 나눠 노출한 뒤, 성과 지표(CTR, CVR, 체류시간 등)를 비교해서 어느 버전이 더 효과적인지를 데이터로 판단하는 실험 방법이다.


1.   무작위 배정이 제대로 되어야 하며(가능하면 유저 단위로 고정), 실험 중 다른 요소(유입채널, 노출시간, 프로모션 등)는 최대한 동일하게 유지해 교란요인을 줄여야 한다.
2.   실험 전에 가설(H0/H1), 메인 지표, 유의수준을 미리 정하고, 중간 결과를 보고 지표를 바꾸거나 유리할 때만 멈추는 행동(피킹)을 피해야 한다.
3. 충분한 표본 수와 기간을 확보해야 하며, 표본이 작으면 우연 변동으로 잘못된 결론이 나올 수 있다.
4. 초두효과(새로움 효과)를 고려해야 한다. 새로운 UI/문구/혜택은 초반에 신기함이나 낯설음 때문에 반응이 일시적으로 과대/과소평가될 수 있으므로, 일정 기간 이상 실험을 진행하거나 초반·후반을 나눠 확인하는 것이 좋다.
5. 결과 해석 시 통계적 유의성(p-value)뿐 아니라 실제 서비스에 의미 있는 개선인지(효과크기)인 실무적 유의성도 함께 판단해야 한다.







---



 A/B TEST를 하기 위해서는 1-pager를 작성해야합니다.  

1-pager란 문제 정의, 관찰하고자하는 실험대상, 가설설정, 평가지표 등을 작성하는  
간단히 말해 계획서를 의미하는데요,

다른 부서와의 소통을 위한 역할 또,과거에 어떤 테스트를
어떻게 진행했는지에 대한 기록의 역할을 하기도 합니다.


 A/B TEST의 1-pager는 복잡한 실험 과정을 한눈에 파악하고, 의사결정권자가 "그래서 이 실험을 왜 했고, 결과가 무엇이며, 무엇을 할 것인가?"를 바로 이해할 수 있도록 작성하는 것이 핵심입니다.

예시를 통해 어떤 항목이 있고, 각 항목에 어떤 내용을 채우는 건지 보여드릴게요.



---



**예시**   
(푸는 거 아니고, 예시입니다! 읽으면서 이런 부분은 이런 내용을 채우는 거군... 이해하시면 돼요!)  




 📊 [ThinkGood] 메인 홈 검색 모듈 상위 노출 실험 1-Pager

---

**1. 실험 전략 및 배경 (Strategy & Observation)**

| 항목 | 내용 |
| :--- | :--- |
| **Target User** | 특정 공모전을 찾으려는 목적이 뚜렷하지만, 현재 하단에 위치한 검색바를 즉각적으로 발견하지 못한 유저 |
| **Observation** | **[데이터 분석 결과]** <br> • 검색 이용 유저의 상세 진입 전환율(CVR)은 40%, 미이용 유저의 전환율은 25%임 <br> • 현재 검색 이용률은 30%로, 검색 가시성 개선을 통한 전환율 상승 여력이 큼 |
| **Problem Statement** | 검색바가 배너 하단에 위치하여 시인성이 낮으며, 유저가 정보를 탐색하기 전 이탈할 확률이 높음 |
| **Hypothesis** | 하단 검색바를 페이지 **최상단으로 노출**시켜 검색 접근성을 높인다면, 전체 공모전 상세 진입률(CVR)이 개선될 것이다 |

---

 **2. 실험군 설계 (Experimental Groups)**

<table>
  <tr>
    <td align="center" width="50%">
      <b>A (Control) - 기존안</b><br>
      <img src="https://i.postimg.cc/kgcLc7Qy/Gemini-Generated-Image-enqvuhenqvuhenqv-(1).png" width="100%"><br>
      <sub>검색바 하단 위치 (기존 UI)</sub>
    </td>
    <td align="center" width="50%">
      <b>B (Variant) - 실험안</b><br>
      <img src="https://i.postimg.cc/0N51Rz5M/Gemini-Generated-Image-enqvuhenqvuhenqv.png" width="100%"><br>
      <sub>검색바 최상단 배치 (변경 UI)</sub>
    </td>
  </tr>
</table>

---

 **3. 지표 및 통계 설계 (Metrics & Statistics)**

> **핵심 측정 지표**
> * **Primary Metric (주지표):** 공모전 상세 페이지 진입률 (Detail Page CVR)
> * **Secondary Metric (보조지표):** 검색바 클릭률 (Search CTR)
> * **Guardrail Metric (방어지표):** 메인 페이지 이탈률 (Bounce Rate)

| 통계적 설계 항목 | 설정값 | 비고 |
| :--- | :--- | :--- |
| **Significance Level ($\alpha$)** | 5% | 유의수준 0.05 |
| **Statistical Power ($1-\beta$)** | 80% | 검정력 유지 |
| **Minimum Sample Size** | **총 92,000명** | 각 그룹당 46,000명 할당 |
| **Required Period** | **최소 4일** |  

---

**4. 위험 관리 및 중단 기준 (Risk & Andon)**

* **Trade-Off:**   
    검색바 상단 노출로 인해 메인 배너의 시각적 주목도가 분산될 수 있으며, 기존 UI에 익숙한 유저에게 일시적인 초두 효과(Primacy Effect)에 의한 혼란이 발생할 수 있음.
* **Andon (중단 기준):**
    1. 대조군 대비 전체 상세 진입률이 **5% 이상 유의미하게 하락**할 경우
    2. UI 변경으로 인한 페이지 오류나 유저 이탈률이 **3일 연속 급증**할 경우

---







<br>

# Q2. 1-pager를 작성해봅시다. (꼬리문제 3 + 보고서 채우는 것1)
저희가 만든 데모 사이트 링크를 첨부해드릴테니, 마음껏 보신 뒤에 위의 예시처럼 1-pager를 작성해보기 바랍니다.

여기서 실험군 설계에 들어가는 사진은 굳이 넣을 필욘 없고 말로만 설명해주셔도 됩니다. 아니면 gemini에게 요청해볼 수도 있겠네요!  

또한 문제탐색에 필요한 cvr과 같은 지표들은 직접 가정하고 하시면 됩니다. 다만, 해당 지표들이 큰 문제가 있는 지표여야... a/b test를 진행하는 이유가 되겠죠!?


정리해보면, 저희 웹사이트 보시고 '이건 여기에 배치하면 더 좋을 거 같은데?' 하는 것들을 찾아서 모든 것을 가정한 뒤에 쓰시면 됩니다!

[데모 웹사이트](https://huggingface.co/spaces/HanJaeHui/ThinkGood_Recommendation_System)


추가적으로 아래 실험군 항목에 사진을 넣으실 분들은 제가 아래 첨부드리는 사이트에 방문하셔서 해당 이미지를 코랩에 넣을 수 있는 형태로 변환한 링크를 복붙하시면 됩니다.  

(.png로 끝나는 링크 복붙하시면 됩니다.)

[포스트 이미지 사이트](https://postimages.org/)



<br>

## Q2-1. 실험군을 설계 해봅시다.
아래 보고서를 채워 주세요.

## A2-1. 아래 보고서 항목 설명을 참고하여 '[ ]' 부분을 지우고 채워주세요.  
  



📊 [프로젝트명] 실험 기획안 1-Pager

---

1. 실험 전략 및 배경 (Strategy & Observation)

| 항목 | 내용 |
| :--- | :--- |
| **Target User** | 공모전을 찾으러 들어온 초기 방문자(특히 비로그인 유저) + 추천을 바로 받고 싶은 유저
 |
| **Observation** | **데이터 근거** <br> • 비로그인 유저 기준, 홈 진입 후 관심분야 필터 사용률이 12%로 낮음(필터 변경 이벤트/세션)  • 비로그인 유저의 공모전 카드 클릭률(CTR)이 3.0%로 낮음(카드 클릭/리스트 노출)  • 세션 리플레이/퍼널 기준(가정), 유저가 첫 행동을 하기 전 좌측 영역을 스캔하지 않고 바로 스크롤하는 비율이 높음 → 필터 UI 발견성이 낮을 가능성

| **Problem Statement** | <br> •  [현재 관심분야 선택 UI가 좌측에 분리되어 있고, 중앙에는 메시지 박스가 크게 배치되어 비로그인 유저가 “바로 필터링”이라는 첫 행동을 찾기 어렵다. 그 결과 필터 사용률이 낮고, 관심사에 맞는 리스트로 좁히지 못해 카드 클릭률이 낮다.
] |
| **Hypothesis** | **[홈 중앙의 메시지 박스를 **필터 UI(관심분야 선택)**로 교체하고, 좌측 라디오를 제거해 필터 기능을 중앙에 집중시키면, 비로그인 유저의 **필터 사용률(12%→20%)**과 **카드 클릭률(3.0%→4.0%)**이 증가할 것이다.

]**  |

---

 2. 실험군 설계 (Experimental Groups)

<table>
  <tr>
    <td align="center" width="50%">
      <b>A (Control) - 기존안</b><br>
      <img src="https://i.postimg.cc/GmwNwNQL/image.png" width="100%"><br>
      <sub>[중앙: “누구나 도전을…” 메시지 박스
	  좌측: 관심분야 라디오 버튼
	  하단: 전체 인기 공모전 리스트]</sub>
    </td>
    <td align="center" width="50%">
      <b>B (Variant) - 실험안</b><br>
      <img src="https://i.postimg.cc/KzTwZs7b/image.png" width="100%"><br>
      <sub>[•중앙 메시지 박스를 필터 박스(관심분야 선택 UI)로 변경
	  좌측 “관심 분야” 라디오 버튼 삭제
	  리스트/카드/CTA 등 나머지는 동일 유지(제가 자체 로컬에서 돌려서 UI가 정확히 똑같지 않은 점은 양해부탁드립니다!😅)
]</sub>
    </td>
  </tr>
</table>


<br>

## Q2-2. 실험에 필요한 최소 test users 수를 구해봅시다.
유의수준 0.05, 검정력 80% 하에 수행해주세요.  


아래 공식 참고해서, 최소 test users 수를 구하는 코드 작성해주세요.

추가적으로 n은 산출하는데에 가정한 p1, p2도 같이 작성해주세요.

$$n = \frac{(Z_{\alpha/2} + Z_{\beta})^2 \cdot [p_1(1-p_1) + p_2(1-p_2)]}{(p_1 - p_2)^2}$$

## A2-2. 아래 질문에 대한 답과 코드를 채워주세요.

p1, p2는 각각 무엇을 의미하는 확률이고, 어떤 값으로 가정하셨나요?   





또한 위 공식을 보고 n을 구하기 위한 코드를 작성해주세요. 단, 라이브러리는 z값을 구하는 용도의 것만 사용해주세요.


### 답변:
*   		p1: **Control(A, 기존안)**에서의 비로그인 유저 기준 주요 지표(여기서는 공모전 카드 클릭률, CTR) 발생 확률
	•	가정: p1 = 0.03 (3.0%)

*       	p2: **Variant(B, 실험안)**에서의 비로그인 유저 기준 공모전 카드 클릭률(CTR) 발생 확률
	•	가정: p2 = 0.04 (4.0%)  (3.0% → 4.0%로 1.0%p 개선 기대)

In [ ]:
# A2-2 코드작성. 나온 n은 아래 Minimum Sample Size에 기재.
import math
from scipy.stats import norm

# given
alpha = 0.05
power = 0.80
beta = 1 - power

# 가정한 p1, p2
p1 = 0.03             # Control의 CTR
p2 = 0.04             # Variant의 CTR

# Z values
z_alpha_2 = norm.ppf(1 - alpha/2)  # 양측검정
z_beta = norm.ppf(1 - beta)

# sample size per group
n = ((z_alpha_2 + z_beta)**2 * (p1*(1-p1) + p2*(1-p2))) / ((p1 - p2)**2)

n_per_group = math.ceil(n)
total_n = 2 * n_per_group

print("Z_alpha/2:", z_alpha_2)
print("Z_beta:", z_beta)
print("n (per group):", n_per_group)
print("total n (A+B):", total_n)

#########

<br>

## Q2-3. 실험에 필요한 최소 실험 기간을 구해봅시다.

이 실험에 필요한 최소 test users가 92000명이 나왔고  
해당 웹페이지 하루 평균 방문자수가 50000명 정도라고 할 때 우리는 최소 4일이 필요합니다.

비교하고 싶은 대상이 총 두개이고, 하나의 대상을 실험하기 위해선 최소 이틀은 필요하기 때문이죠!

위에서 구하신 test users수를 바탕으로, 하루 평균 방문자수를 가정한 뒤 기간을 설정해주세요.  
구하신 기간은 밑에 표에 기재해주시면 됩니다.



## A2-3. 아래 세질문에 답하신 뒤 아래 보고서를 채워주세요.

가정한 하루 평균 방문자 수?  


*  4,000명/일 (비로그인 유저 기준)



산출된 최소 test users 수?

*  10,596명 (A+B 합산)
  
그래서 필요한 최소 기간은?

*  10,596 / 4,000 = 2.649 = 3일
*  다만, 초두효과/요일 효과를 줄이기 위해서 1일 버퍼를 추가해야하니까 최소 4일 운영으로 설정

## **아래 보고서도 채워주세요!**  
**'[ ]' 부분을 지우고 해당 내용에 부합하는 내용들을 채워주시면 됩니다.**

---

**3. 지표 및 통계 설계 (Metrics & Statistics)**

> **핵심 측정 지표**
> * **Primary Metric (주지표):** 공모전 카드 클릭률(CTR) = (공모전 카드 클릭 수) / (공모전 리스트 노출 수)  (비로그인 세션 기준)
> * **Secondary Metric (보조지표):**
    - 필터 사용률(Filter Usage Rate) = (관심분야 필터 변경 세션 수) / (홈 진입 세션 수)
    - 추천 시작 전환율(Start Rate) = (“추천 시작하기” 클릭 세션 수) / (홈 진입 세션 수)
> * **Guardrail Metric (방어지표):**
    - 이탈률(Bounce Rate) = (홈에서 추가 행동 없이 종료한 세션) / (홈 진입 세션)
    - 오류율(Error Rate) = (필터/리스트 로딩 실패, 클릭 이벤트 누락 등) / (홈 진입 세션)

| 통계적 설계 항목 | 설정값 | 비고 |
| :--- | :--- | :--- |
| **Significance Level ($\alpha$)** | 5% | 보통 5%(0.05) 설정 |
| **Statistical Power ($1-\beta$)** | 80% | 보통 80% 설정 |
| **Minimum Sample Size** | **총 10,596명** | 그룹당 인원수 명시 |
| **Required Period** | **최소 4일** | 실험 진행 예정 기간 |

---

 **4. 위험 관리 및 중단 기준 (Risk & Andon)**

* **Trade-Off :**    
    - 기존의 브랜드 메시지(“누구나 도전을…”) 노출이 줄어들어 브랜딩/감성 전달이 약해질 수 있음
	- 필터 UI가 중앙에 오면서 정보가 많아져 첫 화면이 복잡해 보일 위험
	- 좌측 라디오 삭제로 인해 일부 유저는 기존 위치에 익숙했던 탐색 방식이 바뀌어 혼란을 느낄 수 있음

* **Andon (중단 기준) :**
    1.	주지표(공모전 카드 CTR)가 Control 대비 10% 이상 하락(상대 기준)하고, 같은 추세가 연속 1일 이상 지속되면 즉시 중단/롤백
	    •	예: CTR 3.0% → 2.7% 이하

    2.	기술/운영 이슈 발생 시 중단
	    •	필터/리스트 로딩 실패 등 오류율이 2% 이상으로 상승하거나
	    •	“필터가 안 된다/공모전이 안 뜬다” 등 유저 불만(CS/피드백)이 10건 이상/일 발생하면 중단 및 원인 분석 후 재개

---

<br>

# Q3. A/B TEST를 수행해봅시다.

1-pager를 작성했으면 직접 test도 수행해봐야겠죠!  
위에 제가 만들었던 1-pager를 바탕으로 검정을 해볼게요.  
대표적인 검정방법에는 t-test가 있고 카이제곱 겁정이 있습니다.  


---



[실험 가설]  
귀무가설 ($H_0$): 메인 홈 검색 모듈의 위치(하단 vs 최상단)와 상세 페이지 진입 여부 사이에는 상관관계가 없다. 즉, 두 그룹의 상세 페이지 진입률($CVR$)은 동일하다.$$H_0: p_A = p_B$$대립가설 ($H_1$): 메인 홈 검색 모듈을 최상단에 노출하는 것은 상세 페이지 진입 여부에 영향을 미친다. 즉, 실험군(B)의 상세 페이지 진입률이 대조군(A)보다 높을 것이다.$$H_1: p_A < p_B$$


<br>

## Q3-1. 위 가설에 적합한 검정 이름과 왜 해당 검정을 써야하는지 이유도 작성해주세요. (t test와 카이제곱검정 중)

## A3-1.  검정 이름과 이유를 설명해주세요.

답변: 위 가설에 필요한 검정은 위 가설에 필요한 검정은 카이제곱(χ²) 검정 이고,  

이유는 모듈 위치(A/B)도 범주형 변수, 상세 페이지 진입 여부(진입/미진입)도 범주형 변수이므로 두 범주형 변수 간 독립성(연관성) / 비율 차이를 보기 위해  분할표(교차표)를 만들고 **진입률이 동일한지(p_A = p_B)를 **검정하는 상황이기 때문 이다.

<br>

In [ ]:
'''''''''''''''''''''''''''''''''''''''
제가 세운 A/B TEST 설계 실험 코드입니다.
'''''''''''''''''''''''''''''''''''''''

import pandas as pd
from scipy.stats import chi2_contingency

# 1. 관측 데이터 입력 (실제 실험 결과 수치로 변경하세요)
# [상세진입 성공 수, 상세진입 실패 수]
control_results = [11500, 34500]  # 대조군(A)
variant_results = [12190, 33810]  # 실험군(B)

# 2. 분할표(Contingency Table) 생성
data = [control_results, variant_results]

# 3. 카이제곱 검정 수행
chi2, p_value, dof, expected = chi2_contingency(data)

# 4. 결과 출력
print(f"카이제곱 통계량: {chi2:.4f}")
print(f"P-value: {p_value:.10f}")

alpha = 0.05
if p_value < alpha:
    print(f"결과: P-value가 {alpha}보다 작으므로 귀무가설을 기각합니다.")
    print("=> 검색바 최상단 노출이 상세 페이지 진입률에 유의미한 변화를 주었습니다.")
else:
    print(f"결과: P-value가 {alpha}보다 크므로 귀무가설을 기각할 수 없습니다.")
    print("=> 검색바 위치 변경의 효과가 통계적으로 증명되지 않았습니다.")

실제 데이터가 없어서, 저도 필요한 데이터는 가정했답니다.  

이제 25기 분들의 A/B TEST를 수행해보시죠!

<br>

## Q3-2. Q2에서 작성한 실험 수행을 위해 가설을 설정해주세요.


## A3-2. 아래 두 가설 내용을 채워주세요.
  
[실험 가설]  
귀무가설 ($H_0$): A안과 B안의 **상세 페이지 진입률(=공모전 카드 클릭률, CTR)**은 차이가 없다.
  
대립가설 ($H_1$): A안과 B안의 **상세 페이지 진입률(=공모전 카드 클릭률, CTR)**은 차이가 있다.

<br>

## Q3-3. 위 가설을 수행하기 위해 어떤 검정 방법을 사용해야하고 이유는 무엇인지 작성해주세요.

## A3-3. 아래 검정 이름과 이유를 채워주세요.
답변: 위 가설에 필요한 검정은 위 가설에 필요한 검정은 카이제곱(χ²) 검정 이고,

이유는 모듈 위치(A/B)도 범주형 변수, 상세 페이지 진입 여부(진입/미진입)도 범주형 변수이므로 두 범주형 변수 간 독립성(연관성) / 비율 차이를 보기 위해 분할표(교차표)를 만들고 **진입률이 동일한지(p_A = p_B)를 **검정하는 상황이기 때문 이다.

<br>

## Q3-4. 위에서 답한 것을 토대로, 검정을 수행해주세요. (t test와 카이제곱 검정 중)


## A3-4 아래 셀에 코드 작성해주세요. 그리고!!!
그리고 나온 값을 통해 수행하신 a/b test 결과를 설정하신 가설에 맞게 해석해주세요.

In [4]:
### A3-4. 코드작성

import pandas as pd
from scipy.stats import chi2_contingency

# 1. 관측 데이터 입력 (비로그인 유저 기준)
# [카드 클릭(상세진입) 성공 수, 카드 클릭 실패 수]
control_results = [ 159 , 5139 ]   # 대조군(A)
variant_results = [ 212 , 5086 ]   # 실험군(B)

# 2. 분할표(Contingency Table) 생성
data = [control_results, variant_results]

# 3. 카이제곱 검정 수행
chi2, p_value, dof, expected = chi2_contingency(data)

# 4. 결과 출력
print(f"카이제곱 통계량: {chi2:.4f}")
print(f"P-value: {p_value:.10f}")

alpha = 0.05
if p_value < alpha:
    print(f"결과: P-value가 {alpha}보다 작으므로 귀무가설을 기각합니다.")
    print("=> 필터 UI 통합(메인 박스 필터화 & 좌측 라디오 삭제)이 카드 클릭률(상세 진입률)에 유의미한 변화를 주었습니다.")
else:
    print(f"결과: P-value가 {alpha}보다 크므로 귀무가설을 기각할 수 없습니다.")
    print("=> 필터 UI 통합의 효과가 통계적으로 유의하다고 보기 어렵습니다.")


#############

카이제곱 통계량: 7.5529
P-value: 0.0059914921
결과: P-value가 0.05보다 작으므로 귀무가설을 기각합니다.
=> 필터 UI 통합(메인 박스 필터화 & 좌측 라디오 삭제)이 카드 클릭률(상세 진입률)에 유의미한 변화를 주었습니다.


**결과해석**

답변: 카이제곱 독립성 검정 결과 p-value=0.00599로 유의수준 0.05보다 작아 귀무가설을 기각하였다. 따라서 필터 UI 통합 실험안(B)은 대조군(A) 대비 카드 클릭률(3.0%→4.0%)에 유의미한 개선을 가져왔다.

<br>

**1주차 기초통계 과제는 여기까지였습니다. 수고하셨습니다!**